src/main/java/java8/challenges/Encoder.java

```diff
@@ -0,0 +1,61 @@
+package java8.challenges;
+
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Comparator.comparing;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+
+@FunctionalInterface
+public interface Encoder {
+   String encode(String encode) ;
+
+
+    static Encoder getEncoder() {
+      return input -> {
+         Character[] chars = input.chars()
+                 .mapToObj(ch -> (char) ch)
+                 .toArray(Character[]::new);
+
+         List<Character> charList = Arrays.asList(chars);
+
+         Set<Map.Entry<Character, Long>> map = getCharacterLongMap(charList).entrySet().stream().collect(Collectors.toSet());
```

Comment:
can change method name to  something like characterMapWithCount etc . easier to understand on first go .

Reply:
we can't stream an array thats why converting  into an array list

Reply:
Ok , cool